### PR TITLE
Adjusted initial bls12_381 branch

### PR DESCRIPTION
**Pull request by ewynx**
This is a new branch that contains the changes requested in the [previous PR](https://github.com/hashcloak/fuel-crypto/pull/3). 

Changes:

- commits specific for bls12_381 cherry-picked and squashed
- added comments
- updated README
- added licence to `choice.sw`
- created a separate testing folder where all test folder can go into
- moved `choice.sw` into `utils` library so it can be reused easily
- cleanup and changes along the way

This branch works on all comments in the previous PR, except for the last one about assembly versions of the computational heavy operations (whether that would be more efficient than pure Sway). I think it would be better to try that out in a separate branch.

@man2706kum shared that for him there occur some errors in compilation. For me on the other hand; I can compile and run the tests in `testing/tests_bls12_381` for Fp succesfully. Might be some versioning issue, would be interesting to know whether you can compile @Mikerah.

### CODE REVIEW

File: testing/tests_bls12_381/tests/tests_fp/mod.rs [hunk 1]
```diff
@@ -0,0 +1,231 @@
+use fuels::{
+    prelude::*,
+    tx::{ConsensusParameters, ContractId},
+};
+
+abigen!(BlsTestContract, "out/debug/tests_bls12_381-abi.json");
+
+//TODO move this to a separate helpers file. How to achieve this in Rust/Cargo?
```
Comment (Mikerah) [inline testing/tests_bls12_381/tests/tests_fp/mod.rs:8]: Use Rust modules e.g. `helpers/mod.rs` where `helpers` is some other folder you create. Once you have enough helpers, you can further refactor that into its own modules.

**Review by Mikerah — APPROVED** [review]
We should clearly not what tests run but take way too long to run. Otherwise, LGTM

**Comment by ewynx** [comment]
Made the last fixes as suggested, will merge now

**Comment by Mikerah** [comment]
:rocket:

**Pull request merged**